app/services/api/v1/ecf/induction_records_query.rb

```diff
@@ -13,40 +13,41 @@ def all
       induction_records = relevant_induction_records
 
       if updated_since.present?
-        induction_records = induction_records.where("induction_records.updated_at > ?", updated_since)
+        # this has to check all updatable entities
+        induction_records = induction_records
+                              .joins(participant_profile: [:user])
+                              .where("induction_records.updated_at > ? OR participant_profiles.updated_at > ? OR users.updated_at > ?", updated_since, updated_since, updated_since)
       end
 
       if email.present?
+        # this has to return just one record
         induction_records = induction_records.joins(:preferred_identity).where(preferred_identity: { email: })
       end
 
       induction_records
     end
 
     def relevant_induction_records
-      InductionRecord
-        .joins(
-          <<-SQL,
-            JOIN (#{induction_record_history.to_sql}) AS historical_induction_records
-              ON historical_induction_records.id = induction_records.id AND historical_induction_records.chronology = 1
-            JOIN participant_profiles ON induction_records.participant_profile_id = participant_profiles.id
-          SQL
-        )
-        .merge(ParticipantProfile.ecf)
+      InductionRecord.where(id: induction_record_history)
     end
 
     def induction_record_history
-      InductionRecord.start_date_in_past
-                     .select(
-                       <<-SQL,
-                         induction_records.id,
-                         ROW_NUMBER() OVER (
-                           PARTITION BY induction_records.participant_profile_id
-                           ORDER BY induction_records.end_date ASC NULLS FIRST, induction_records.start_date ASC
-                         ) AS chronology
-                       SQL
-                     )
-                     .joins(:participant_profile)
+      # we can't use `InductionRecord.current` here, because we need to include records that finished in the past if there is no current one
+      # we can't use `InductionRecord.latest` here because the latest record might not be what they are doing right now
+      #
+      # we need to find the oldest `participant_profile` for the user otherwise we risk unwanted changes to their access
+      # we need to prioritise `end_date` with `null` first because we need to find the record describing what they are doing right now over what they will eventually do after a transfer
+      # we need to order by `start_date` second in case two records have the same `end_date` such as after modifying a transfer to add a mentor
+      InductionRecord.select(
+        <<-SQL,
+          FIRST_VALUE(induction_records.id) OVER (
+            PARTITION BY "users"."id"
```

Comment:
You could probably replace with `teacher_profile.id` and save a join?

Reply:
even simpler to use `"participant_profiles"."teacher_profile_id"` and reduce to one join that already has the right constraints on it.